File: Iteration1/oosd_project/src/main/java/com/cse3063f25grp1/orchestrator/SequentialRagPipeline.java
package com.cse3063f25grp1.orchestrator;

import com.cse3063f25grp1.config.Config;
import com.cse3063f25grp1.context.Context;

public class SequentialRagPipeline extends RagPipeline {

    public SequentialRagPipeline(Config config, Context context, com.cse3063f25grp1.trace.TraceBus traceBus) {
        super(config, context, traceBus);
    }

    @Override
    public void execute() {

        detectIntent();
        writeQuery();
        retrieve();
        rerank();
        answer();
    }
}

File: iteration2/src/orchestrator/sequential_rag_pipeline.py
from src.orchestrator.rag_pipeline import RagPipeline
from src.config.config import Config
from src.context.context import Context
from src.trace.trace_bus import TraceBus


class SequentialRagPipeline(RagPipeline):
  
    def __init__(self, config: Config, context: Context, trace_bus: TraceBus):
       
        super().__init__(config, context, trace_bus)
    
    def execute(self) -> None:
       
        self.detect_intent()
        self.write_query()
        self.retrieve()
        self.rerank()
        self.answer()
